components/base/MediaItem.vue

```diff
@@ -174,7 +176,13 @@ const toggleContent = () => {
 
 const isMediaItemHovering = useElementHover(mediaItem)
 
-defineExpose({ isLewdBlurredLayer })
+function toggleVideoFullscreen() {
+  if (video.value.toggleVideoFullscreen) {
+    video.value.toggleVideoFullscreen()
+  }
+}
```

Comment:
It makes no sense to call `toggleVideoFullscreen` inside the common `MediaItem` component. 

I would make the change like below:
MediaItem.vue:  `mediaRef.value.toggleFullscreen() `
VideoMedia.vue:  `const toggleFullscreen = () => { video.value.toggleVideoFullscreen()}`

Reply:
I don't think it should follow this way. `toggleVideoFullscreen` is a method exposed directly from  `VideoMedia`. 

```
videomedia
const { toggle: toggleVideoFullscreen } = useFullscreen(video)

defineExpose({ toggleVideoFullscreen })
```
Calling this `const toggleFullscreen = () => { video.value.toggleVideoFullscreen()}` in the `MediaItem` component is because I created a `ref` called `video` on the `component` tag. 

```
<template>
  <div ref="mediaItem" class="media-object h-fit">
    <component
      :is="resolveComponent"
      ref="video"
      
      ....
      
      ````

Reply:
I'm not talking about whether it will work. The current implementation will work but it's not easy to extend in the future. It also looks confusing. Setting `video` ref inside `MediaItem` is improper because it assumes that this component is only for `video`. 

What if I would like to customize the fullscreen process for `image` in the current implementation pattern? We need a unified method to adopt fullscreen mode for different kinds of nft.

Reply:
Yes, absolutely, I agree with you. The unified method for fullscreen mode is actually already set in the `GalleryItem` component. And I think the reason each media type wasn't given its own fullscreen method initially was to avoid unnecessary duplication. However, it appears video is an exception. The current component hierarchy has a ref introduced for at least all child components used in each of the module files. The `useFullscreen` api requires the direct ref of the video file provided, and if otherwise, might not work as desired, which explains the issue with toggling video files to fullscreen. 

Perhaps, a trade-off here might be to use a generic name for the ref other than `video`. To reduce confusion.

Reply:
> However, the `useFullscreen` api requires the direct ref of the video file provided, and if otherwise, might not work as desired, which explains the issue with toggling video files to fullscreen.
> 

Could you please check this? It should work.
https://github.com/kodadot/nft-gallery/pull/10089#pullrequestreview-2020074089